dotnet/src/SemanticKernel.Abstractions/Security/TrustAwareString.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Security;
+
+/// <summary>
+/// A string wrapper that carries trust information.
+/// All field are readonly.
+/// </summary>
+public class TrustAwareString
```

Comment:
Note for follow-up PR:
TrustAwareString: IEquatable<TrustAwareString>, IEquatable<string>

Reply:
Oh yes, specially if we update it to be a struct.